shell/src/main/java/alluxio/cli/fsadmin/report/MetricsCommand.java

```diff
@@ -0,0 +1,129 @@
+  private FileSystemMasterClient mFileSystemMasterClient;
+  private PrintStream mPrintStream;
+
+  /**
+   * Creates a new instance of {@link MetricsCommand}.
+   *
+   * @param fileSystemMasterClient client to connect to filesystem master client
+   * @param printStream stream to print operation metrics information to
+   */
+  public MetricsCommand(FileSystemMasterClient fileSystemMasterClient, PrintStream printStream) {
+    mFileSystemMasterClient = fileSystemMasterClient;
+    mPrintStream = printStream;
+    mCachedRpcInvocation = new StringBuilder();
+  }
+
+  /**
+   * Runs report metrics command.
+   *
+   * @return 0 on success, 1 otherwise
+   */
+  public int run() throws IOException {
+    Map<String, Long> metricsMap = new TreeMap<>(mFileSystemMasterClient.getMetrics());
+    metricsMap.put("FilesPinned", metricsMap.get("master.FilesPinned"));
+
+    Set<String> operations = new HashSet<>();
+    operations.add("DirectoriesCreated");
+    operations.add("FileBlockInfosGot");
+    operations.add("FileInfosGot");
+    operations.add("FilesCompleted");
+    operations.add("FilesCreated");
+    operations.add("FilesFreed");
+    operations.add("FilesPersisted");
+    operations.add("FilesPinned");
+    operations.add("NewBlocksGot");
+    operations.add("PathsDeleted");
+    operations.add("PathsMounted");
+    operations.add("PathsRenamed");
+    operations.add("PathsUnmounted");
+
+    Set<String> rpcInvocations = new HashSet<>();
+    rpcInvocations.add("CompleteFileOps");
+    rpcInvocations.add("CreateDirectoryOps");
+    rpcInvocations.add("CreateFileOps");
+    rpcInvocations.add("DeletePathOps");
+    rpcInvocations.add("FreeFileOps");
+    rpcInvocations.add("GetFileBlockInfoOps");
+    rpcInvocations.add("GetFileInfoOps");
+    rpcInvocations.add("GetNewBlockOps");
+    rpcInvocations.add("MountOps");
+    rpcInvocations.add("RenamePathOps");
+    rpcInvocations.add("SetAttributeOps");
+    rpcInvocations.add("UnmountOps");
+
+    mPrintStream.println("Alluxio logical operations: ");
+    for (Map.Entry<String, Long> entry : metricsMap.entrySet()) {
+      String key = entry.getKey();
+      if (operations.contains(key)) {
+        // Print operation info first
+        printOrCache(key, entry.getValue(), false);
+      } else if (rpcInvocations.contains(key)) {
+        // Cache RPC invocation info and print later
+        printOrCache(key, entry.getValue(), true);
+      }
+    }
+
+    mPrintStream.println("\nAlluxio RPC invocations: ");
+    mPrintStream.println(mCachedRpcInvocation.toString());
+    // TODO(lu) provide other kind metrics information
+    return 0;
+  }
+
+  /**
+   * Prints or caches indented metrics information of a certain metrics property.
+   *
+   * @param key the key of the metrics property
+   * @param value the value of the metrics property
+   * @param cache true to cache and print later, false to print directly
+   */
+  private void printOrCache(String key, Long value, boolean cache) {
```

Comment:
it is unnecessarily trick to treat operation info and rpc info differently (cache or not), is the benefit just to save one pass? I would not compromise code readability to achieve such optimization --- this is on the shell (client) side, we should be good even we pay one more pass.

Reply:
Make sense! I changed to two passes and improve code readability! Thanks